src/test/CharterManager-unit.t.sol

```diff
@@ -386,6 +378,60 @@ contract CharterManagerTest is TestBase {
         assertEq(a.gems(), 100 * 1e18);
     }
 
+    function test_exit() public {
+        (Usr a,) = init_user();
+        assertEq(gem.balanceOf(address(a)), 200 * 1e6);
+
+        a.join(200 * 1e6);
+        assertEq(a.gems(), 200 * 1e18);
+        assertEq(gem.balanceOf(address(a)), 0);
+
+        // check exit of unlocked gems does not affect the vault and does not prevent increasing debt
+        a.exit(200 * 1e6);
+        assertEq(a.gems(), 0);
+        assertEq(gem.balanceOf(address(a)), 200 * 1e6);
+    }
+
+    function test_exit_other() public {
+        (Usr a, Usr b) = init_user();
+        assertEq(gem.balanceOf(address(a)), 200 * 1e6);
+
+        a.join(200 * 1e6);
+        assertEq(a.gems(), 200 * 1e18);
+        assertEq(gem.balanceOf(address(a)), 0);
+
+        a.hope(address(b));
+        b.flux(address(a), address(b), 200 * 1e18);
+        b.exit(address(a), 200 * 1e6);
+        assertEq(a.gems(), 0);
+        assertEq(b.gems(), 0);
+        assertEq(gem.balanceOf(address(a)), 200 * 1e6);
+    }
+
+    function testFail_exit_other() public {
+        (Usr a, Usr b) = init_user();
+        assertEq(gem.balanceOf(address(a)), 200 * 1e6);
+
+        a.join(200 * 1e6);
+        assertEq(a.gems(), 200 * 1e18);
+        assertEq(gem.balanceOf(address(a)), 0);
+
+        // b can not exit a's funds
+        b.exit(address(a), 200 * 1e6);
```

Comment:
Hmm I don't understand this statement. Here fails because `b` doesn't have funds to send that amount whatever it is to itself or to `a`.
I think this test might be a bit legacy from when you had extended the `exit` function before rolling it back.

Reply:
Right, removed.